### PR TITLE
HPCC-20539 Escape "$" to preserve variable for error handling

### PR DESCRIPTION
**Pull request by xwang2713**
Change log message from "WARNING" to "FATAL_ERROR" to stop the build if error happens

Signed-off-by: Xiaoming Wang <xiaoming.wang@lexisnexi.com> 

@kunalaswani please review

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.

## Testing:

Build OK in normal case: 
   http://10.240.32.243/job/CE-JIRA-HPCC-20539/CE=centos-7.0-x86_64/2/console
Build failed in npm error:
   http://10.240.32.243/job/CE-JIRA-HPCC-20539/CE=centos-7.0-x86_64-2/4/console

### CODE REVIEW

**Comment by kunalaswani** [comment]
@xwang2713 The changes look good.

**Comment by xwang2713** [comment]
@richardkchapman this is ready to merge. 
Without the fix platform build will succeed even though npm has build error. At run-time user will run problem with ECLWatch.

File: esp/src/CMakeLists.txt [hunk 1]
```diff
@@ -62,21 +62,21 @@ else ()
                 message ( \"---- Installing third party libraries\" )
                 execute_process ( WORKING_DIRECTORY \"${ECLWATCH_BUILD_TMP}\" COMMAND \"npm\" \"install\" \"--loglevel=error\" \"--prefer-offline\" OUTPUT_FILE \"${ECLWATCH_BUILD_INSTALL_OUT}\" RESULT_VARIABLE _BUILD_RESULT )
                 if (_BUILD_RESULT)
-                    message ( WARNING \"npm install failed see above and ${ECLWATCH_BUILD_INSTALL_OUT} for details.\" )
+                    message ( FATA_ERROR \"npm install failed see above and ${ECLWATCH_BUILD_INSTALL_OUT} for details.\" )
```
Comment (richardkchapman) [inline esp/src/CMakeLists.txt]: Should say FATAL_ERROR I imagine...

File: esp/src/CMakeLists.txt [hunk 2]
```diff
@@ -45,8 +45,8 @@ else ()
                 set (NODE_ERROR \"Unable to locate node/npm, please install NodeJS as per https://github.com/hpcc-systems/HPCC-Platform/wiki/Building-HPCC#prerequisites\" )
             endif ()
 
-            if ( NOT \"${NODE_ERROR}\" STREQUAL \"\")
-                message ( WARNING \"${NODE_ERROR}\" )
+            if ( NOT \"\${NODE_ERROR}\" STREQUAL \"\")
```
Comment (richardkchapman) [inline esp/src/CMakeLists.txt]: I don't understand why the " and the $ need escaping here, but if they do then they presumably do a couple of lines later too...

Comment (richardkchapman) [reply esp/src/CMakeLists.txt]: Actually I think I do understand, and it's probably ok in the other places (will just expand earlier, but that's ok for the other cases).

**Comment by richardkchapman** [comment]
@xwang2713 One typo

**Comment by xwang2713** [comment]
Changed "FATA_ERROR" to "FATAL_ERROR". Thanks

**Comment by richardkchapman** [comment]
Doesn't look like you have pushed that last change

**Review by xwang2713 — COMMENTED** [review]
Done.

**Pull request merged**